pkg/commands/plan/plan_init.go

```diff
@@ -0,0 +1,243 @@
+}
+
+func (c *PlanInitCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Initialize Guardian for running the Terraform plan process.
+`
+}
+
+func (c *PlanInitCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet(c.testFlagSetOpts...)
+
+	c.GitHubFlags.AddFlags(set)
+	c.RetryFlags.AddFlags(set)
+
+	f := set.NewSection("Command options")
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "dest-ref",
+		Target:  &c.flagDestRef,
+		Example: "ref-name",
+		Usage:   "The destination GitHub ref name for finding file changes.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-ref",
+		Target:  &c.flagSourceRef,
+		Example: "ref-name",
+		Usage:   "The source GitHub ref name for finding file changes.",
+	})
+
+	f.StringSliceVar(&cli.StringSliceVar{
+		Name:    "directories",
+		Target:  &c.flagDirectories,
+		Example: "dir1,dir2,dir3",
+		Usage:   "The subset of directories to target for Terraform planning.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "delete-outdated-comments",
+		Target:  &c.flagDeleteOutdatedComments,
+		Default: true,
+		Usage:   "Delete outdated comments when Guardian plan is run multiple times for the same pull request.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "json",
+		Target:  &c.flagJSON,
+		Default: false,
+		Usage:   "Print the output directories in JSON format.",
+	})
+
+	return set
+}
+
+func (c *PlanInitCommand) Run(ctx context.Context, args []string) error {
+	logger := logging.FromContext(ctx)
+
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	c.workingDir = cwd
+
+	c.gitClient = git.NewGitClient(cwd)
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithInitialRetryDelay(c.RetryFlags.FlagInitialRetryDelay),
+		github.WithMaxRetries(c.RetryFlags.FlagMaxRetries),
+		github.WithMaxRetryDelay(c.RetryFlags.FlagMaxRetryDelay),
+	)
+
+	if len(c.flagDirectories) == 0 {
+		c.flagDirectories = append(c.flagDirectories, cwd)
+	}
+
+	logger.Debugln("finding entrypoint directories..")
+
+	dirs, err := terraform.GetEntrypointDirectories(c.flagDirectories)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+
+	c.entrypoints = dirs
+	logger.Debugf("terraform entrypoint directories: %s", c.entrypoints)
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian plan init process.
+func (c *PlanInitCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
```

Comment:
I'd probably add a bunch of context to the logger here:

```suggestion
	logger := logging.FromContext(ctx).
		Named("plan.process").
		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
		With("github_repo", ...)
```